kratos/includes/checks.h

```diff
@@ -46,6 +46,47 @@ KRATOS_ERROR << "The string \"" << SubString << "\" was not found in the given s
 " is not near to " << #b << " = " << b << " within the tolerance " << tolerance
 #define KRATOS_CHECK_DOUBLE_EQUAL(a,b) KRATOS_CHECK_NEAR(a,b,std::numeric_limits<double>::epsilon())
 
+#define KRATOS_CHECK_VECTOR_NEAR(a, b, tolerance) {                        \
+KRATOS_ERROR_IF_NOT(a.size() == b.size())                                  \
+<< "Check failed because vector arguments do not have the same size:"      \
+<< std::endl                                                               \
+<< "First argument has size " << a.size() << ", "                          \
+<< "second argument has size " << b.size() << "." << std::endl;            \
+for (std::size_t i = 0; i < a.size(); i++) {                               \
+   KRATOS_ERROR_IF( std::abs(a[i] - b[i]) > tolerance )                    \
+   << "Check failed because vector " << #a << " with values" << std::endl  \
+   << a << std::endl                                                       \
+   << "Is not near vector " << #b << " with values" << std::endl           \
+   << b << std::endl                                                       \
+   << "Mismatch found in component " << i << ":" << std::endl              \
+   << a[i] << " not near " << b[i]                                         \
+   << " within tolerance " << tolerance << "." << std::endl;               \
+}                                                                          \
+}
+#define KRATOS_CHECK_VECTOR_EQUAL(a, b) KRATOS_CHECK_VECTOR_NEAR(a,b,std::numeric_limits<double>::epsilon())
+
+#define KRATOS_CHECK_MATRIX_NEAR(a, b, tolerance) {                              \
+KRATOS_ERROR_IF_NOT((a.size1() == b.size1()) || (a.size2() == b.size2()))        \
```

Comment:
I found a library which replaces the error based tests for a inner check, appparently the compilation is way faster. For the future @KratosMultiphysics/technical-committee : https://github.com/onqtam/doctest

![imagen](https://user-images.githubusercontent.com/19991680/63529116-75e51800-c504-11e9-81b4-1672d8780b35.png)